### PR TITLE
Update cfn-init template so it is functional with a modern Ubuntu 20.04 LTS

### PR DESCRIPTION
**Pull request by witchbutter**
## Description of the Change

The current template references old AMIs for Ubuntu 16.04 which is deprecated.  This template is modernized for Ubuntu 20.04 and cfn-init 2.0 which supports Python 3.4 and later. It was manually tested and validated.

## Possible Drawbacks

The only drawback is whether the intent of the exercise is to get aws-cfn-bootstrap functional on Ubuntu before actually moving on to the topic of the log config

### CODE REVIEW

**Review by zyjia8 — APPROVED** [review]

File: 08-cloudwatch-logs/8.1.2.yml [hunk 1]
```diff
@@ -1,100 +1,85 @@
...
           Name: !Ref InstanceProfile
         NetworkInterfaces:
-          -
-            DeviceIndex: 0
+          - DeviceIndex: 0
             AssociatePublicIpAddress: True
-            SubnetId: !FindInMap [ RegionMap, !Ref "AWS::Region", subnetId ]
-        UserData: 
-          Fn::Base64: !Sub | 
+            SubnetId: #Fill in your subnet ID
```
Comment (arothian) [inline 08-cloudwatch-logs/8.1.2.yml:19]: I think this probably just makes sense as another template parameter.

File: 08-cloudwatch-logs/8.1.2.yml [hunk 2]
```diff
@@ -1,100 +1,85 @@
...
+            Action:
               - "sts:AssumeRole"
   SecurityGroup:
     Type: AWS::EC2::SecurityGroup
-    Properties: 
-      GroupDescription: Topic 5 - EC2 Lab
-      VpcId: !FindInMap [ RegionMap, !Ref "AWS::Region", vpcId ]
+    Properties:
+      GroupDescription: 'Stelligent-U 08 CloudWatch'
+      VpcId: #Fill in your VPC ID
```
Comment (arothian) [inline 08-cloudwatch-logs/8.1.2.yml:85]: Ditto, would make this a template parameter as well.

File: 08-cloudwatch-logs/8.1.2.yml [hunk 3]
```diff
@@ -3,6 +3,10 @@ Parameters:
   UbuntuAMI:
     Type: "AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>"
     Default: '/aws/service/canonical/ubuntu/server/20.04/stable/current/amd64/hvm/ebs-gp2/ami-id'
+  VPCID:
+    Type: String
+  SubnetID:
+    Type: String
```
Comment (arothian) [inline 08-cloudwatch-logs/8.1.2.yml:9]: Recommend using a stronger type matched to the expected input
```suggestion
    Type: 'AWS::EC2::Subnet::Id'
```

File: 08-cloudwatch-logs/8.1.2.yml [hunk 4]
```diff
@@ -3,6 +3,10 @@ Parameters:
   UbuntuAMI:
     Type: "AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>"
     Default: '/aws/service/canonical/ubuntu/server/20.04/stable/current/amd64/hvm/ebs-gp2/ami-id'
+  VPCID:
+    Type: String
```
Comment (arothian) [inline 08-cloudwatch-logs/8.1.2.yml:7]: Recommend using a stronger type constraint
```suggestion
    Type: 'AWS::EC2::VPC::Id'
```

**Review by arothian — APPROVED** [review]

**Pull request merged**